woocommerce/woocommerce-gutenberg-products-block
# Show attributes in simple products and variable products

In `wc/store/v1/products` we return product attributes for both simple and variable products, but we had a bug in our logic in which we never return attributes unless those attributes are both used for variations and are visible, in reality, we only want to show attributes that are **either** visible, **or** has variations, or both.

It's not directly obvious from the code that we support simple product attributes, but given the logic in this line

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/66c1b505c4ad00abae9e5729c420248552ef3d56/src/StoreApi/Schemas/V1/ProductSchema.php#L683

It makes sense that we will return both types of attributes.

Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/6180

### Manual Testing

How to test the changes in this Pull Request:

1. Create a simple product with attributes and set them as visible.
2. Using a REST client, fetch `wc/store/v1/products/:id` and check the attributes field for the product where `:id` is the product id, in both cases, you should see the attributes you added.
3. Uncheck visible attributes, fetch your product again, no attributes should be returned.
4. Turn the product into a variable product, keep attributes hidden but check "used for variations".
5. Fetch your product again, you should see your attributes.
6. Make them visible, you should still see the attributes.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).




### Changelog

> Store API: show visible attributes in simple products, and hidden attributes in variable products.
